tests/fault_tolerance/cancellation/test_sglang.py

```diff
@@ -0,0 +1,398 @@
+                    "Chat completion stream request cancellation",
+                ),
+            ]
+
+            for i, (request_type, description) in enumerate(test_scenarios, 1):
+                logger.info(f"Testing {description.lower()}...")
+                send_request_and_cancel(request_type)
+
+                logger.info(
+                    "Checking for cancellation messages in worker and frontend logs..."
+                )
+                time.sleep(1)  # time for cancellation to propagate
+                frontend_log_offset, worker_log_offset, _ = verify_request_cancelled(
+                    frontend,
+                    worker,
+                    frontend_log_offset=frontend_log_offset,
+                    worker_log_offset=worker_log_offset,
+                )
+
+                logger.info(f"{description} detected successfully")
+
+
+@pytest.mark.e2e
+@pytest.mark.sglang
+@pytest.mark.gpu_1
+@pytest.mark.model(FAULT_TOLERANCE_MODEL_NAME)
+def test_request_cancellation_sglang_prefill_cancel(
+    request, runtime_services, predownload_models
+):
+    """
+    End-to-end test for request cancellation during prefill phase.
+
+    This test verifies that when a request is cancelled by the client during the prefill phase,
+    the system properly handles the cancellation and cleans up resources
+    on the prefill worker side in a disaggregated setup.
+    """
+
+    # Step 1: Start the frontend
+    with DynamoFrontendProcess(request) as frontend:
+        logger.info("Frontend started successfully")
+
+        # Step 2: Start the decode worker
+        logger.info("Starting decode worker...")
+        decode_worker = DynamoWorkerProcess(request, mode="decode")
+
+        with decode_worker:
+            logger.info(f"Decode Worker PID: {decode_worker.get_pid()}")
+
+            # Step 3: Start the prefill worker
+            logger.info("Starting prefill worker...")
+            prefill_worker = DynamoWorkerProcess(request, mode="prefill")
+
+            with prefill_worker:
+                logger.info(f"Prefill Worker PID: {prefill_worker.get_pid()}")
+
+                # TODO: Why wait after worker ready fixes frontend 404 / 500 flakiness?
+                time.sleep(10)
+
+                # Step 4: Test request cancellation during prefill phase
+                logger.info(
+                    "Testing completion request cancellation during prefill phase..."
+                )
+                send_request_and_cancel("completion", timeout=1, use_long_prompt=True)
+
+                logger.info(
+                    "Checking for cancellation messages in prefill and decode worker and frontend logs..."
+                )
+                time.sleep(1)  # time for cancellation to propagate
+                verify_request_cancelled(
+                    frontend,
+                    decode_worker,
+                    prefill_worker,
+                    assert_cancel_at_remote_worker=False,
```

Comment:
```suggestion
                    assert_cancel_at_remote_worker=True,
```
We should set this to `True` to verify cancellation log is printed at the remote worker, in this case, it is the remote prefill worker.